ci/upload_to_owncloud.sh

```diff
@@ -137,12 +137,29 @@ main() {
     local_path="$1"; shift
 
     info "upload $local_path to $OWNCLOUD_BROWSE_URL/$remote_path/$(basename "$local_path") (using user $user)"
+
+    if ! command -v uuidgen > /dev/null ; then
```

Comment:
So now we always upload to a temporary location before moving it to the final location? Isn't it wastefull? Uploading a full logs diretocy of 20+ tests takes A LOT of time as it is, I think this feature should be optional unless you tested it with a big folder with lots of files and found out that the move part is significantly faster than the upload part.

Reply:
Yes we always upload to a temporary location first, then do a move.

If the destination folder doesn't exist the move is instant since it's just a rename, so there shouldn't be any significant overhead.

When the folder already exists however (and it does for the `latest` folder), it takes much more time.
If we didn't upload to a temporary folder first, we would anyway have to remove the destination folder first before uploading, so I'm not convinced it would take less time.


To summarize: 
I think the move is annoying for the `latest` folder, because I assumed it would almost instantly replace the folder (we want it to contain valid artifacts at any time) and it doesn't, but it has nothing to do with the fact that we use temporary folders.
To improve this we could try [Frederik's suggestion](https://github.com/prplfoundation/prplMesh/pull/1050#discussion_r400861726) and see if we can use some kind of link instead of a plain folder.

Reply:
I like Frederik's suggestion. So with that, we could say that we *require* the target folder not to exist, right?

That said, I think we should merge it as is now, even if the copy takes some time, and fix later. For Tomer's use case the temporary location shouldn't hurt, and this way we at least have something working for the "latest".

Reply:
With a link, we would be able to make it point to something that is valid at any point in time, so it won't matter much anymore if destination folder already exist or not.

I agree it should be good enough for now.